### DISCUSSION TITLE
Raspberry Pi supported?

### DISCUSSION THREAD

**Discussion by dhajoglou** [electronstudio/jaylib#29] [Q&A] [answered]
I tried to run the 3.7 jars  (4.0 jars seem to be missing from the release section) and got an error about linux-armhf:
Exception in thread "main" java.lang.UnsatisfiedLinkError: Platform "linux-armhf" not supported by class com.raylib.Raylib

So, I thought i'd try to build on Raspberry Pi.  I'm getting this error when starting the process.  What am I missing?

root@wm-media:~/code/git/jlib# ./build-java.sh 
STEP 1 - generate Raylib.java
Info: javac -cp ../javacpp.jar:/root/code/git/jlib/src com/raylib/RaylibConfig.java 
Info: Targeting ../gen/com/raylib/Raylib.java
Info: Nothing targeted for ../gen/com/raylib/Raylib.java
STEP 2 - compile Raylib.java
Info: javac -cp ../javacpp.jar:/root/code/git/jlib/gen com/raylib/Raylib.java 
error: file not found: com/raylib/Raylib.java
Usage: javac <options> <source files>
use --help for a list of possible options
Exception in thread "main" java.lang.RuntimeException: Could not compile com/raylib/Raylib.java: 2
	at org.bytedeco.javacpp.tools.Builder.main(Builder.java:1430)
Fix this before trying again

===
Linux wm-media 5.10.63-v7+ #1496 SMP Wed Dec 1 15:58:11 GMT 2021 armv7l GNU/Linux
java javac 11.0.12
openjdk version "11.0.12" 2021-07-20
OpenJDK Runtime Environment (build 11.0.12+7-post-Raspbian-2deb10u1)
OpenJDK Server VM (build 11.0.12+7-post-Raspbian-2deb10u1, mixed mode)

root@wm-media:~/code/git/jlib# ls -la /usr/local/lib/
total 3284
drwxr-xr-x  4 root root     4096 Feb 10 18:15 .
drwxr-xr-x 10 root root     4096 Jan 28 02:24 ..
-rw-r--r--  1 root root  1691846 Feb 10 18:14 libraylib.a
lrwxrwxrwx  1 root root       16 Feb 10 18:15 libraylib.so -> libraylib.so.400
-rwxr-xr-x  1 root root  1647788 Feb 10 18:15 libraylib.so.4.0.0
lrwxrwxrwx  1 root root       18 Feb 10 18:15 libraylib.so.400 -> libraylib.so.4.0.0

**Comment by electronstudio** [comment]
Reason there are no jars on GitHub are they are in Maven repo now. But in any case you will have to compile your own. 

Which Pi are you using? I have a 3 I could test on but not a 4. 

I think the problem is probably that JavaCPP doesn't have a configuration entry for ARM platform so will need to add one.,,

**Reply by electronstudio** [reply]
Yeah will need an entry made in src/com/raylib/RaylibConfig.java I think.

**Comment by electronstudio** [comment, marked as answer]
i've done a build for my rapsberry pi 3, and uploaded all the jars to the releases section, including the new `jaylib-natives-linux-armhf-4.0.0.jar`

There is also a new build script, `build-linux-armhf.sh`

Does this work on your raspberry pi?

**Reply by dhajoglou** [reply]
I am using a PI3 with the Raspberry Pi Os: Buster. 

I'll give this a shot tomorrow both with your jar and attempting to build it myself as well.  Thanks for the fast feedback!

My project is for streaming media titles for whitewater kayak racing.  I prototyped it with a simple framebuffer library.  That worked so moving up to some more featured and java-ed solutions!

**Reply by dhajoglou** [reply]
I re-pulled the master branch and ran the build-linux-armhf.sh script.  It built it and the build process ran the test in com/raylib/Test folder successfully!  I then repeated the test with my own quick mod an it worked like a charm!

Thanks for the addition for Raspberry Pi 3!

**Reply by electronstudio** [reply]
That's good.  Note that on the Pi3 it used to be possible to get better performance by using 'native' opengl video chip drivers that don't use X11.  But they changed how this works, so I'm not sure if the X11 drivers are now as good, and I'm not sure if Raylib is still able to compile with the native mode option or not.
